### ISSUE TITLE
[Grid - Basics] Internal accessibility audit

### ISSUE THREAD

**Issue by DanielleRameau** [sl-design-system/components#2500]
### Provide a general summary of the issue here

Test all basics of grid on accessibility and create new issues in GitHub when there are issues encountered.
Newly made a11y issues will be added to this milestone: https://github.com/sl-design-system/components/milestone/23

### 👤 Your name

DR

**Comment by KatarzynaWM** [comment]
testing environment and tools:

Windows 11
Chrome 
screen reader: NVDA 2025.3
Color Contrast Analyser from TPGi

---------
If you have a narrow screen or increase zoom, the tooltip appears that shows the text content that is too long to fit in the cell. This tooltip is accessible only via mouse.

On mobile the tooltip is visible only when you keep your finger on the screen.

<img width="713" height="511" alt="Image" src="https://github.com/user-attachments/assets/3b3e06c2-99b9-4bc1-bbf4-64a6c854c316" />

\
👩‍💻 It's something to fix (the tooltip should be available for keyboard users as well), @DanielleRameau can you please make an issue for this one?

👨‍🔬 This issue changed a bit, can't be reproduced as it was, but similar issue still exist and is reported as #3271

**Comment by KatarzynaWM** [comment]
Sometimes, when zooming in, instead of tooltips the second horizontal scroll bar appear. This makes the table very difficult to navigate. This happens also on mobile.

<img width="1879" height="636" alt="Image" src="https://github.com/user-attachments/assets/252c973a-7240-423e-99c6-2dcd26b9f3c4" />


\
👩‍💻 We need to investigate zooming in/out in different browsers for web components, Jeroen already added an issue #2825 for it, but maybe it's worth to add this comment (with screenshot) as a comment there or sub-issue, @DanielleRameau?

👨‍🔬 In several independent tests, this issue could not be reproduced. Similar issue still exist and is reported as #3271

**Comment by KatarzynaWM** [comment]
Border color of the grid has to low contrast ratio to the background (1,4:1 - should be at least 3:1).

**Comment by KatarzynaWM** [comment]
First row has `aria-rowindex="0"`. According to [documentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-rowindex): "The value for `aria-rowindex` is an integer greater than or equal to 1, greater than the `aria-rowindex` value of any previous rows, and less than or equal to the number of rows in the full table." 

<img width="1804" height="438" alt="Image" src="https://github.com/user-attachments/assets/89dfa0eb-ccd7-4cf1-8a86-174f1ad6a1fd" />

**Comment by KatarzynaWM** [comment]
[Column Groups]
Columns ("first name", "last name" etc.). are not gruped correctly under first level headers ("name", "school").  "First name" element is announced correctly, but "Last name" is being associated with "school" instead of "name", so screen reader announces e.g. "School Last Name column 2 Muller". Check this for possible solutions: https://www.w3.org/WAI/tutorials/tables/irregular/#table-with-two-tier-headers

**Comment by KatarzynaWM** [comment]
[Header]

Information in the tooltip are available only for mouse users:

<img width="442" height="151" alt="Image" src="https://github.com/user-attachments/assets/f991265a-8afc-4899-829e-c1117cc303fa" />

**Comment by a11ymiko** [comment]
Issues were created for all findings from the audit.
